### PR TITLE
chore: fixes and updates keycloak development environment

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Updates the `realm-export.json` with defaults for OpenID:
  - Created a `mattermost-openid` client inside the existing `mattermost` realm
- Updated the `README.md` file with all relevant configuration, information and made it more readable overall.
- Added a new `make config-openid` that sets up the `config.json` using a file with the default settings and `jq`.

```release-note
chore: improvements to openid local development
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Nice work!

File: server/build/docker-compose.common.yml [hunk 1]
```diff
@@ -104,17 +104,17 @@ services:
...
     restart: always
+    command: start-dev --import-realm --verbose
     environment:
-      KEYCLOAK_USER: mmuser
-      KEYCLOAK_PASSWORD: mostest
-      DB_VENDOR: h2
-      KEYCLOAK_IMPORT: /setup/realm.json
+      KEYCLOAK_ADMIN: mmuser
+      KEYCLOAK_ADMIN_PASSWORD: mostest
+      KC_DB: dev-file
```
Comment (hanzei) [inline server/build/docker-compose.common.yml:113]: What does this env variable do?

**Review by hanzei — COMMENTED** [review]
The changes look fine. Thanks for updating the image!

Just as a heads up: @BenCookie95 did a very similar PR via https://github.com/mattermost/mattermost/pull/26313. Do we need both PRs? I quite like the docs update from this PR, but on the other hand `make config-saml` is handy.

File: server/build/docker-compose.common.yml (hunk 1 above)
Comment (fmartingr) [reply server/build/docker-compose.common.yml:113]: [Selects the database mode for keycloak](https://www.keycloak.org/server/db)

**Comment by fmartingr** [comment]
> The changes look fine. Thanks for updating the image!
> 
> Just as a heads up: @BenCookie95 did a very similar PR via #26313. Do we need both PRs? I quite like the docs update from this PR, but on the other hand `make config-saml` is handy.

I guess we can merge relevant changes into his PR if that's being worked on. 

> but on the other hand make config-saml is handy.

I agree is super useful, but one reason I removed the JSON from the docs (apart from being outdated) is that I wanted to test that it worked from the UI to understand how it works. It not a PITA since I only did it once, but the command may be useful for ephemeral environments, though I'm not sure how many people uses this since it has been outdated for a while.

**Comment by hanzei** [comment]
> I agree is super useful, but one reason I removed the JSON from the docs (apart from being outdated) is that I wanted to test that it worked from the UI to understand how it works. It not a PITA since I only did it once, but the command may be useful for ephemeral environments, though I'm not sure how many people uses this since it has been outdated for a while.

Sound good to have both: the command and the written instructions :+1:

**Review by streamer45 — APPROVED** [review]
Nice!

**Comment by fmartingr** [comment]
> > I agree is super useful, but one reason I removed the JSON from the docs (apart from being outdated) is that I wanted to test that it worked from the UI to understand how it works. It not a PITA since I only did it once, but the command may be useful for ephemeral environments, though I'm not sure how many people uses this since it has been outdated for a while.
> 
> Sound good to have both: the command and the written instructions 👍

I will wait then until his PR is merged and then adjust mine accordingly.

**Comment by hanzei** [comment]
@fmartingr Heads up that https://github.com/mattermost/mattermost/pull/26313 got merged.

**Comment by fmartingr** [comment]
> @fmartingr Heads up that #26313 got merged.

Hey @hanzei, sorry for the delay! I've updated the PR accordingly. One change I made is using `jq` to populate the default OpenID settings. I believe this to be better (we all should have jq installed anyway) so we have a file with the default settings with is more readable and easier to iterate on, `sed` has limitations.

Let me know what you think, I can either move my changes back to use `sed` or change the SAML one to use `jq` as well.

File: server/Makefile [hunk 2]
```diff
@@ -691,6 +691,19 @@ config-saml: ## Configures SAML.
...
+config-openid: ## Configures OpenID.
+	@echo Setting up configuration for local OpenID with keycloak, please ensure your keycloak is running on http://localhost:8484
+
+	# Check if jq is installed
+	@jq --version > /dev/null 2>&1 || (echo "jq is not installed. Please install jq to continue." && exit 1)
+
+	TMPDIR=$(mktemp -d)
+	jq --slurp '.[0] * .[1]' config/config.json build/docker/keycloak/openid.mmsettings.json > ${TMPDIR}/config.json
+	cp ${TMPDIR}/config.json ./config/config.json
+	rm ${TMPDIR}/config.json
```
Comment (hanzei) [inline server/Makefile:703]: Nice use of `jq` :+1: Do you also want to migrate `config-saml` and `config-ldap` as part of this PR?

**Review by hanzei — APPROVED** [review]
Awesome, love to see the use of OpenID :rocket:

**Review by agnivade — APPROVED** [review]

File: server/Makefile (hunk 2 above)
Comment (fmartingr) [reply server/Makefile:703]: I was about to do it but I saw that `config-saml` and `config-ldap` use `../config/config.json` for the config file path, while my environment uses `config/config.json`. Which one is correct? I want to unify usage. cc: @BenCookie95

Comment (fmartingr) [reply server/Makefile]: https://github.com/mattermost/mattermost/pull/26518/commits/d78d662890677c26f66e84fafd82b9bd01a5a957

**Comment by hanzei** [comment]
Sorry for the long delay @fmartingr! @BenCookie95 would you mind taking a look at the PR as you worked don't he keycloak setup before?

**Review by BenCookie95 — APPROVED** [review]
This is great, thanks Felipe

**Pull request merged**